homeassistant/components/kostal_plenticore/coordinator.py

```diff
@@ -59,7 +59,16 @@ async def async_setup(self) -> bool:
             async_get_clientsession(self.hass), host=self.host
         )
         try:
-            await self._client.login(self.config_entry.data[CONF_PASSWORD])
+            if (
+                "Master Key" in self.config_entry.data
+                and "Service Code" in self.config_entry.data
+            ):
+                await self._client.login(
+                    self.config_entry.data["Master Key"],
+                    service_code=self.config_entry.data["Service Code"],
+                )
+            else:
+                await self._client.login(self.config_entry.data[CONF_PASSWORD])
```

Comment:
As I mentioned in config flow, this if-condition should not be necessary at all, if key is used instead of password.